### PR TITLE
feat: add esm support

### PR DESCRIPTION
**Pull request by TrySound**
In this diff I added esm support with the help of cherry-pick package
and added babel helpers.

### CODE REVIEW

**Comment by TrySound** [comment]
Hi @jquense. Pinging you in case you missed this :)

**Comment by jquense** [comment]
If the tree shaking works i'd be fine with a single main export version and forgo the cherry-pick tooling since it's really just two public imports here the HOC and hook.

**Comment by TrySound** [comment]
Cool. Yeah, treeshaking will work with only functions.

**Comment by TrySound** [comment]
Your babel config produces this things which should be development only
image

**Comment by jquense** [comment]
I'll take a look, i think it's jsut a configuration issue. need to update the preset

**Comment by TrySound** [comment]
Got it. Your config requires `NODE_ENV=production` everywhere.

File: README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ import uncontrollable from 'uncontrollable'
 For every prop you indicate as uncontrollable, the returned component will also accept an initial, `default` value for that prop. For example, `open` can be left uncontrolled but the initial value can be set via `defaultOpen={true}` if we want it to start open.
 
 ```js
-const uncontrollable = require('uncontrollable')
+import { uncontrollable } from 'uncontrollable)
```
Comment (jquense) [inline README.md]:
```suggestion
import { uncontrollable } from 'uncontrollable'
```

**Comment by TrySound** [comment]
@jquense Could you publish new version?

**Comment by TrySound** [comment]
Friendly ping @jquense :)

**Comment by TrySound** [comment]
@jquense Ooops. It should be major release. We did breaking change.
import uncontrollable from 'uncontrollable' -> import { uncontrollable } from 'uncontrollable'
import useUncontrolled from 'uncontrollable/hooks' -> import { useUncontrolled } from 'uncontrollable'

**Comment by jquense** [comment]
I (hopefully) tweaked build output to avoid the breaking change. you can still cherry-pick `useUncontrolled` from `uncontrolled/hook` it just won't work correctly in the esm case, which should be ok since we didn't have esm support before?

**Comment by TrySound** [comment]
Ah, cool. Didn't notice :)

**Comment by TrySound** [comment]
@jquense Though you didn't provide default export for uncontrollable. So this is still breaking change.

**Comment by TrySound** [comment]
I suggest to rerelease v6.1.0 and release master as v7

**Comment by chengjia0807** [comment]
same issue here

**Comment by TrySound** [comment]
@chengjia0807 Try uncontrollable 6.2.2

**Comment by jquense** [comment]
sorry folks, should be good now.

**Pull request merged**